### ISSUE TITLE
Embeddings after tiling process

### ISSUE THREAD

**Issue by maiskovich** [prov-gigapath/prov-gigapath#52]
Hello, 

Thank you for sharing the code for such a great project.
I am getting embeddings that are not meaningful and I wanted to check if I might be doing something that is not what the model is expecting. 

I am running the gigapath tile encoder on tiles that I generated through a custom pipeline where I discard the background and only generate tiles that have enough content. I am testing this on tissue microarrays cores which are relatively small. For example, for one core, after removing background, I end with 14 tiles, each being 512x512.

When I run the tile encoder, after saving the embeddings as pytorch tensors, I get as an output a total of 14 files, summing up to 11 mb. As I remove the background, my coordinates are from the only parts of the images where we have content, for example, I have tiles that start from X: 0 Y: 922, or X: 1383, Y 1383, where many parts of the image are not covered (there was only background in that section). 

Then, I run the slide encoder with these images, and I get back, for the whole core, an embedding that when saved as a pytorch tensor is only 42kb. 

Is the slide encoder expected to deal with situations with background removal and incomplete coordinates or should I just pass all the tiles to it (including pure background ones)? Are the file sizes that I am getting expected? the slide embedding for me looks kind of small, but wanted to double check and I might be wrong on what is expected

Thank you

**Comment by HanwenXuTHU** [comment]
Hi, thanks for reaching out! The output of the slide encoder is the [CLS] token and that is used for finetuning the slide encoder on downstream tasks, e.g., classifications. That's why the file size seems to be smaller.

**Comment by maiskovich** [comment]
Okay, thank you!

So, should this tiling strategy of avoiding the background be fine with the gigapath embedding? Could you please confirm the coordinate format?

For downstream tasks, in the slide embedding, I have the following keys (as explained in the demo in this repository) 'layer_0_embed', 'layer_1_embed', 'layer_2_embed', 'layer_3_embed', 'layer_4_embed', 'layer_5_embed', 'layer_6_embed', 'layer_7_embed', 'layer_8_embed', 'layer_9_embed', 'layer_10_embed', 'layer_11_embed', 'layer_12_embed', 'last_layer_embed'

Which layer is recommended to use for downstream tasks? Or if the idea is to use multiple ones, what is the recommended strategy for that?



Thanks!

**Comment by HanwenXuTHU** [comment]
Hi sorry for the late response! I determine the layers to use based on a hold-out validation set. Yes, you're right, using multiple ones could be better than single layer.
